### PR TITLE
PLUGIN-496: Support syntax checking in BigQuery execute plugin.

### PR DESCRIPTION
**Pull request by jasir99**
Support syntax checking in BigQuery execute plugin.

JIRA Ticket: https://cdap.atlassian.net/browse/PLUGIN-496

### CODE REVIEW

File: src/test/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecuteConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    FieldSetter.setField(config, GCPConfig.class.getDeclaredField("project"), "test_project");
+
+    MockFailureCollector failureCollector = new MockFailureCollector();
+    BigQuery bigQuery = mock(BigQuery.class);
+    when(bigQuery.create(ArgumentMatchers.any(JobInfo.class))).thenThrow(new BigQueryException(1, errorMessage));
+
+    config.validateSQLSyntax(failureCollector, bigQuery);
+    Assert.assertEquals(1, failureCollector.getValidationFailures().size());
+    Assert.assertEquals(String.format("%s.", errorMessage),
+                        failureCollector.getValidationFailures().get(0).getMessage());
```
Comment (flakrimjusufi) [inline src/test/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecuteConfigTest.java:66]: We can also try to catch the config property in which we are expecting the validation to fail.

File: src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java [hunk 2]
```diff
@@ -274,5 +291,27 @@ public void validate(FailureCollector failureCollector) {
...
+      QueryJobConfiguration queryJobConfiguration = QueryJobConfiguration.newBuilder(sql).setDryRun(true).build();
+      try {
+        bigQuery.create(JobInfo.of(queryJobConfiguration));
+      } catch (BigQueryException e) {
+        failureCollector.addFailure(String.format("%s.", e.getMessage()), "Please specify a valid query.")
+          .withConfigProperty(SQL);
+      }
+    }
+
+    private BigQuery getBigquery(FailureCollector failureCollector) {
```
Comment (flakrimjusufi) [inline src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java]: nit: We can change the method name to something like `getBigQuery`

**Review by flakrimjusufi — CHANGES_REQUESTED** [review]

File: src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java [hunk 3]
```diff
@@ -152,7 +154,7 @@ public AbstractBigQueryActionConfig getConfig() {
   /**
    * Config for the plugin.
    */
-  public final class Config extends AbstractBigQueryActionConfig {
+  public static class Config extends AbstractBigQueryActionConfig {
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java]: why did we make this class not final?

File: src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java [hunk 4]
```diff
@@ -253,8 +261,17 @@ public void validate(FailureCollector failureCollector) {
         }
       }
 
-      if (!containsMacro(SQL) && Strings.isNullOrEmpty(sql)) {
-        failureCollector.addFailure("SQL not specified. Please specify a SQL to execute", null).withConfigProperty(SQL);
+      if (!containsMacro(SQL)) {
+        if (Strings.isNullOrEmpty(sql)) {
+          failureCollector.addFailure("SQL not specified. Please specify a SQL to execute", null)
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java]: failureCollector.addFailure("SQL not specified.",  "Please specify a SQL to execute.").withConfigProperty(SQL);

File: src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java (hunk 3 above)
Comment (jasir99) [reply src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java]: PR updated

File: src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java (hunk 4 above)
Comment (jasir99) [reply src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java]: PR updated

**Review by rmstar — APPROVED** [review]
lgtm, please squash commits

**Comment by rmstar** [comment]
Tests are failing with this error:
```
Error: No commit found for SHA: PLUGIN_496/bigquery_execute_syntax_checking
```
@jasir99 can you please rebase against latest develop branch? I think that would fix it.

**Comment by jasir99** [comment]
> Tests are failing with this error:
> 
> ```
> Error: No commit found for SHA: PLUGIN_496/bigquery_execute_syntax_checking
> ```
> 
> @jasir99 can you please rebase against latest develop branch? I think that would fix it.

@rmstar we rebased our branch against latest develop branch and we are still facing the same issue. Any suggestion on how to proceed?

**Comment by rmstar** [comment]
Not sure why it's failing - looks unrelated to your changes. I'll merge this PR and look into the failure separately.

**Pull request merged**